markdowns/19_Text_to_Image_search_pipeline_with_MultiModal_Retriever.md

```diff
@@ -0,0 +1,187 @@
+last_updated: 2022-11-01
+level: "intermediate"
+weight: 95
+description: Use a MultiModalRetriever to build a cross modal search pipeline.
+category: "QA"
+aliases: ['/tutorials/multimodal']
+---
+    
+
+**Level**: Intermediate
+
+**Time to complete**: 20 minutes
+
+**Prerequisites**: Prepare the Colab environment (see links below).
+
+**Nodes Used**: InMemoryDocumentStore, MultiModalRetriever
+
+**Goal**: After completing this tutorial, you will have learned about the MultiModalRetriever, and built a simple retrieval pipeline that search relevant images given a text query.
+
+## Preparing the Colab Environment
+
+- [Enable GPU Runtime in GPU](https://docs.haystack.deepset.ai/v5.2-unstable/docs/enable-gpu-runtime-in-colab)
+- [Check if GPU is Enabled](https://docs.haystack.deepset.ai/v5.2-unstable/docs/check-if-gpu-is-enabled)
+- [Set logging level to INFO](https://docs.haystack.deepset.ai/v5.2-unstable/docs/set-the-logging-level)
+
+
+```bash
+%%bash
+
+pip install --upgrade pip
+pip install farm-haystack[colab]
+```
+
+
+```python
+import logging
+
+logging.basicConfig(format="%(levelname)s - %(name)s -  %(message)s", level=logging.WARNING)
+logging.getLogger("haystack").setLevel(level=logging.INFO)
+```
+
+# Initializing the DocumentStore
+
+A DocumentStore contains Documents, which in this case are references to the images that Haystack will compare with your query. Here we are using the InMemoryDocumentStore since it requires no external dependencies. To learn more about the DocumentStore and the different types of external databases that we support, see [DocumentStore](https://docs.haystack.deepset.ai/docs/document_store).
+
+
+```python
+from haystack.document_stores import InMemoryDocumentStore
+
+document_store = InMemoryDocumentStore(embedding_dim=512)
+```
+
+# Downloading data
+
+Download 18 sample images of different animals from . You can find them in data/tutorial19/spirit-animals/ as a set of .jpg files.
+
+
+```python
+from haystack.utils import fetch_archive_from_http
+
+doc_dir = "data/tutorial19"
+
+fetch_archive_from_http(
+    url="https://s3.eu-central-1.amazonaws.com/deepset.ai-farm-qa/datasets/documents/spirit-animals.zip",
+    output_dir=doc_dir
+)
+```
+
+Add the images you just downloaded into Haystack Document objects and write them into the DocumentStore.
+
+
+```python
+import os
+
+from haystack import Document
+
+images = [
+    Document(content=f"./{doc_dir}/spirit-animals/{filename}", content_type="image")
+    for filename in os.listdir(f"./{doc_dir}/spirit-animals/")
+]
+
+document_store.write_documents(images)
+```
+
+# Initializing the Retriever
+
+Retrievers sift through all the images and return only those that are relevant based on the input query. Here we are using the OpenAI CLIP model to embed images. For more details on supported modalities, see [MultiModalRetriever](https://docs.haystack.deepset.ai/docs/retriever#multimodal-retrieval).
```

Comment:
Is there model card for the OpenAI CLIP model? Would be good to link it here if so

Reply:
Added.